### ISSUE TITLE
docker:publishLocal fails with missing dependency for com.advancedtelematic:libats_2.12:0.4.0-20-ge903ac1

### ISSUE THREAD

**Issue by bpsrinivas** [simao/ota-lith#2]
Build fails with the following trace

[info] set current project to ota-lith (in build file:/C:/work/15_Connected_Services/01_OTA/ota-lith/)
[info] Wrote C:\work\15_Connected_Services\01_OTA\ota-lith\target\scala-2.12\ota-lith_2.12-c9251e8dc45fac82b585c74976b3f048ecc90cbb.pom
[warn]
[warn]  Note: Unresolved dependencies path:
[warn]
[warn]  Note: Unresolved dependencies path:
[error] sbt.librarymanagement.ResolveException: Error downloading com.advancedtelematic:libats_2.12:0.4.0-20-ge903ac1
[error]   Not found
[error]   Not found
[error]   not found: C:\Users\pbalakr3\.ivy2\localcom.advancedtelematic\libats_2.12\0.4.0-20-ge903ac1\ivys\ivy.xml
[error]   not found: https://repo1.maven.org/maven2/com/advancedtelematic/libats_2.12/0.4.0-20-ge903ac1/libats_2.12-0.4.0-20-ge903ac1.pom
**[error]   not found: https://nexus.ota.here.com/content/repositories/releases/com/advancedtelematic/libats_2.12/0.4.0-20-ge903ac1/libats_2.12-0.4.0-20-ge903ac1.pom**
[error]   not found: https://nexus.ota.here.com/content/repositories/snapshots/com/advancedtelematic/libats_2.12/0.4.0-20-ge903ac1/libats_2.12-0.4.0-20-ge903ac1.pom
[error]   not found: https://version99.qos.ch/com/advancedtelematic/libats_2.12/0.4.0-20-ge903ac1/libats_2.12-0.4.0-20-ge903ac1.pom
[error]         at lmcoursier.CoursierDependencyResolution.unresolvedWarningOrThrow(CoursierDependencyResolution.scala:258)
[error]         at lmcoursier.CoursierDependencyResolution.$anonfun$update$38(CoursierDependencyResolution.scala:227)
[error]         at scala.util.Either$LeftProjection.map(Either.scala:573)
[error]         at lmcoursier.CoursierDependencyResolution.update(CoursierDependencyResolution.scala:227)
[error]         at sbt.librarymanagement.DependencyResolution.update(DependencyResolution.scala:60)
[error]         at sbt.internal.LibraryManagement$.resolve$1(LibraryManagement.scala:53)
[error]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$12(LibraryManagement.scala:103)
[error]         at sbt.util.Tracked$.$anonfun$lastOutput$1(Tracked.scala:73)
[error]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$20(LibraryManagement.scala:116)
[error]         at scala.util.control.Exception$Catch.apply(Exception.scala:228)
[error]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11(LibraryManagement.scala:116)
[error]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11$adapted(LibraryManagement.scala:97)
[error]         at sbt.util.Tracked$.$anonfun$inputChangedW$1(Tracked.scala:219)
[error]         at sbt.internal.LibraryManagement$.cachedUpdate(LibraryManagement.scala:130)
[error]         at sbt.Classpaths$.$anonfun$updateTask0$5(Defaults.scala:3486)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error]         at sbt.Execute.work(Execute.scala:291)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
[error]         at java.base/java.lang.Thread.run(Thread.java:832)
[error] sbt.librarymanagement.ResolveException: Error downloading com.advancedtelematic:libats-http_2.12:0.4.0-20-ge903ac1
[error]   Not found
[error]   Not found
[error]   not found: C:\Users\pbalakr3\.ivy2\localcom.advancedtelematic\libats-http_2.12\0.4.0-20-ge903ac1\ivys\ivy.xml
[error]   not found: https://repo1.maven.org/maven2/com/advancedtelematic/libats-http_2.12/0.4.0-20-ge903ac1/libats-http_2.12-0.4.0-20-ge903ac1.pom
[error]   not found: https://nexus.ota.here.com/content/repositories/releases/com/advancedtelematic/libats-http_2.12/0.4.0-20-ge903ac1/libats-http_2.12-0.4.0-20-ge903ac1.pom
[error]   not found: https://nexus.ota.here.com/content/repositories/snapshots/com/advancedtelematic/libats-http_2.12/0.4.0-20-ge903ac1/libats-http_2.12-0.4.0-20-ge903ac1.pom
[error]   not found: https://version99.qos.ch/com/advancedtelematic/libats-http_2.12/0.4.0-20-ge903ac1/libats-http_2.12-0.4.0-20-ge903ac1.pom

**Comment by simao** [comment]
Hi,

Please see my comment here: https://github.com/simao/ota-lith/issues/1#issuecomment-876257288

You will need to checkout that commit and publishLocal before building.

**Comment by simao** [comment]
Now that all changes are merged upstream in uptane (see github.com/uptane/), this project now uses the official repositories as upstream.

I pulled all changes and we are currently up to date with master.

This means you no longer need to do the manual libats publishing and `dokcer:publishLocal` should work now to build an ota-lith container.

Also, I pushed an image to `uptane/ota-lith` so you don't need to build a docker image if you don't want to, just use `uptane/ota-lith:<version>`

Where version is a commit from master. `git rev-parse master` for example.
